### DISCUSSION TITLE
Unable to create view summarizing video count by category in PocketBase

### DISCUSSION THREAD

**Discussion by intansemc2** [pocketbase/pocketbase#4433] [Q&A] [answered]
I'm trying to create a view in PocketBase that summarizes the number of videos for each category. 
I have two tables:
+ videos: This table contains information about videos. It has a multiple relation field named "categories".
+ categories: This table stores information about the video categories. It has three column: name, description and thumbnail

The closest solution i come up with is create a view with query:

```
SELECT videos.id as id, videos.categories as categories
FROM videos 
```

and use Javascript to calculate number of videos for each categories.

I appreciate any guidance or suggestions for creating the desired view in PocketBase.

**Comment by ganigeorgiev** [comment, marked as answer]
Multiple `relation` field values are stored as serialized json array so you'll need to use the SQLite [`json_each` function](https://www.sqlite.org/json1.html).

Based on the description of your tables, one possible query could be (_I haven't tested it_):

```sql
SELECT
    categories.id,
    categories.name,
    count(videos.id) as totalVideos
FROM categories
LEFT JOIN videos on EXISTS(
    SELECT 1 FROM JSON_EACH(videos.categories) je WHERE je.value = categories.id
)
GROUP BY categories.id
```

**Reply by intansemc2** [reply]
Thank you so much! That worked perfectly!
